src/main/java/hmim/eteam/rest/backend/api/IRoleResolver.java

```diff
@@ -0,0 +1,15 @@
+package hmim.eteam.rest.backend.api;
+
+import hmim.eteam.rest.backend.entity.user.AuthToken;
+import hmim.eteam.rest.backend.entity.user.SiteUser;
+import hmim.eteam.rest.backend.entity.user.UserRole;
+import org.springframework.lang.Nullable;
+
+import javax.validation.constraints.NotNull;
+import java.util.Optional;
+
+public interface IRoleResolver {
+    UserRole resolve(@NotNull String authTokenId, @Nullable Long courseId);
```

Comment:
As I remember, we already receiving `Long` from autogenerated REST interface layer, so there will be no allocations. I don't think reserving invalid good in this case, because it will require database id generation mechanism change and will not achieve anything, because long is already allocated and either way we'll compare it with reserved invalid value (null or proposed constant).